The `bps_tree.perftest` warns on compiling

```
[100%] Building CXX object perf/CMakeFiles/memtx.perftest.dir/memtx.cc.o
/__w/luajit/luajit/perf/bps_tree.cc:172:26: warning: offset of on non-standard-layout type 'DummyAllocator<8192>' [-Winvalid-offsetof]
  172 |                 DummyAllocator *self = container_of(allocator, DummyAllocator,
      |                                        ^
  173 |                                                     matras_allocator);
      |                                                     ~~~~~~~~~~~~~~~~
/__w/luajit/luajit/src/trivia/util.h:297:29: note: expanded from macro 'container_of'
  297 |         (type *)( (char *)__mptr - offsetof(type,member)  );})
      |                                    ^             ~~~~~~
/usr/lib/llvm-19/lib/clang/19/include/__stddef_offsetof.h:16:24: note: expanded from macro 'offsetof'
   16 | #define offsetof(t, d) __builtin_offsetof(t, d)
      |                        ^                     ~
/__w/luajit/luajit/perf/bps_tree.cc:149:19: note: in instantiation of member function 'DummyAllocator<8192>::alloc' requested here
  149 |                                         EXTENT_SIZE, alloc, free);
      |                                                      ^
/__w/luajit/luajit/perf/bps_tree.cc:230:1: note: in instantiation of member function 'DummyAllocator<8192>::DummyAllocator' requested here
  230 | CREATE_TREE_CLASS(tree_i64);
      | ^
/__w/luajit/luajit/perf/bps_tree.cc:205:5: note: expanded from macro 'CREATE_TREE_CLASS'
  205 |                 : Allocator_Base(count * sizeof(elem_t) * 2) \
      |                   ^
gmake[3]: Leaving directory '/__w/luajit/luajit'
[100%] Built target tuple_format.test
```

@ligurio PTAL as an original author.

--- a/perf/bps_tree.cc
+++ b/perf/bps_tree.cc
@@ -138,7 +138,7 @@
  */
 template<int EXTENT_SIZE>
 struct DummyAllocator {
-	std::unique_ptr<char[]> m_buf;
+	char *m_buf;
 	size_t m_buf_size;
 	size_t m_pos = 0;
 	struct matras_allocator matras_allocator;
@@ -152,12 +152,13 @@ struct DummyAllocator {
 		/* The calculated size is incorrect for small trees. */
 		if (m_buf_size < EXTENT_SIZE * 10)
 			m_buf_size = EXTENT_SIZE * 10;
-		m_buf = std::unique_ptr<char[]>(new char[m_buf_size]);
+		m_buf = new char[m_buf_size];
 	}
 
 	~DummyAllocator()
 	{
 		matras_allocator_destroy(&matras_allocator);
+		delete[] m_buf;
 	}
 
 	void

